src/Microsoft.DotNet.Wpf/src/WindowsBase/MS/Internal/AvTrace.cs

```diff
@@ -194,53 +198,35 @@ void Initialize( )
         //  the TraceSource.)
         //
 
-        static private bool ShouldCreateTraceSources()
+        private static bool ShouldCreateTraceSources(bool isEnabledInRegistry)
         {
-            if( IsWpfTracingEnabledInRegistry()
-                || IsDebuggerAttached()
-                || _hasBeenRefreshed
-              )
-            {
-                return true;
-            }
-
-            return false;
+            return isEnabledInRegistry || IsDebuggerAttached() || _hasBeenRefreshed;
         }
 
-
-
-        ///
-        ///  Read the registry to see if WPF tracing is allowed
-        ///
-
-        [FriendAccessAllowed]
-        static internal bool IsWpfTracingEnabledInRegistry()
+        /// <summary> Initializes <see cref="s_enabledInRegistry"/> variable from Registry and returns the value. </summary>
+        /// <returns> A boolean value specifying whether WPF Tracing is enabled. </returns>
+        internal static bool LoadWpfTracingSettings()
         {
-            // First time this is called, initialize from the registry
-
-            if( _enabledInRegistry == null )
-            {
-                bool enabled = false;
+            // Initialize from the registry
+            bool enabled = false;
 
-                object keyValue = SecurityHelper.ReadRegistryValue(
-                                                            Registry.CurrentUser,
-                                                            @"Software\Microsoft\Tracing\WPF",
-                                                            "ManagedTracing");
+            object keyValue = SecurityHelper.ReadRegistryValue(Registry.CurrentUser,
+                                                               @"Software\Microsoft\Tracing\WPF",
+                                                               "ManagedTracing");
 
-                if( keyValue is int && ((int) keyValue) == 1 )
-                {
-                    enabled = true;
-                }
+            if (keyValue is int value && value == 1) //REG_DWORD
+                enabled = true;
 
-                // Update the static.  Doing this last protects us from threading problems; worse case, multiple
-                // threads will set the same value into it.
-                _enabledInRegistry = enabled;
-            }
+            // Update the static value
+            s_enabledInRegistry = enabled;
 
-            return (bool) _enabledInRegistry;
+            return enabled;
         }
 
-
+        /// <summary> Retrieves cached value from <see cref=" s_enabledInRegistry"/> to see whether WPF Tracing is enabled. </summary>
+        /// <returns> A boolean value specifying whether WPF Tracing is enabled. </returns>
+        [FriendAccessAllowed]
+        internal static bool IsWpfTracingEnabledInRegistry() => s_enabledInRegistry;
```

Comment:
Can it sure already call the `LoadWpfTracingSettings` method to init the `s_enabledInRegistry` field?

Reply:
@lindexi It is initialized by the static constructor, line 60. So the field will have the correct value once this is called.